fitbenchmarking/fitbenchmarking
# Pylint - Fail on refactor and convention warnings

#### Description of Work
This PR is an attempt to squash all refactor and convention warnings seen in the pylint output. The motivation for doing this is so that it is easier to read the failures that relate to your PR. It also ensures unwanted refactor or convention practises do not get into the code base.

Fixes #957

#### Testing Instructions

1. Ensure the pylint job passes. Read the output of the pylint and make sure there are no convention or refactor warnings.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
